### PR TITLE
Moved some validations to the serializer

### PR DESCRIPTION
**Pull request by marianoeramirez**
@anx-cbenke 
I created a new pull request with the same changes from the old pull request. because some changes were not intended to merge on this request.  

These changes are only related to moving some of the validations to the serializer.

### CODE REVIEW

File: django_rest_passwordreset/views.py [hunk 1]
```diff
@@ -160,7 +127,7 @@ def post(self, request, *args, **kwargs):
         if not active_user_found and not getattr(settings, 'DJANGO_REST_PASSWORDRESET_NO_INFORMATION_LEAKAGE', False):
             raise exceptions.ValidationError({
                 'email': [_(
-                    "There is no active user associated with this e-mail address or the password can not be changed")],
+                    "We couldn't find an account associate with that email. Please try a different e-mail address.")],
```
Comment (anx-cbenke) [inline django_rest_passwordreset/views.py:130]: associate**d** ;-)

**Comment by anx-hnezbeda** [comment]
@marianoeramirez I would sugest to do an [Interactive Rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History) and squash the 7 commits into 1, this will make the history cleaner,

To fix the merge conflict you can rebase afterwards you change with the master branch. Make sure you do a force push on your branch afterwards or else git will reject the modification because of the change in the history.

**Comment by marianoeramirez** [comment]
@anx-hnezbeda I am not sure if I did the squash right. I squash the commits into c7542f9. 

Should I delete the old commits?

**Comment by anx-hnezbeda** [comment]
@marianoeramirez you can try to follow this guide: https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git

At the end you will have only 1 commit in your PR as all changes will be combined to one commit. Currently there are 11. Let me know if you need any more help with this :slightly_smiling_face:

As mentioned before, you will need to to a force push to your branch as this is a change in the git history.

**Comment by marianoeramirez** [comment]
@anx-hnezbeda I think now yes it's ok. let me know otherwise

**Comment by anx-hnezbeda** [comment]
@marianoeramirez This looks much better :slightly_smiling_face:, but there is one thing left. We merged some pull requests over the past days and this probably causes some conflicts on your side, which you have probably solved in the merge commit that was generated. To avoid this you can do a rebase with the `master` branch of the source repository. Here are the required steps:

* Add the source remote: `git remote add upstream https://github.com/anexia-it/django-rest-passwordreset.git`
* Fetch the recent changes merged into source: `git fetch upstream`
* Remove the merge commit: `git reset --hard HEAD~1`
* Rebase with the source `master` branch: `git rebase upstream/master`
* Fix potential merge conflicts and continue rebase `git rebase --continue` (you will need this only if you have any conflicts during the rebase: https://docs.github.com/en/github/using-git/resolving-merge-conflicts-after-a-git-rebase)
* Force push your changes: `git push -f origin move_serializer`

Here is a gist with some more details for such an approach: https://gist.github.com/ravibhure/a7e0918ff4937c9ea1c456698dcd58aa

**Comment by marianoeramirez** [comment]
I think now yes is only one commit.

**Comment by anx-hnezbeda** [comment]
@marianoeramirez looks great :+1:

**Pull request merged**